src/environmentd/src/http/sql.rs

```diff
@@ -53,11 +56,75 @@ struct ErrorResponse {
     error: String,
 }
 
-pub async fn handle_sql_ws(client: AuthedClient, ws: WebSocketUpgrade) -> impl IntoResponse {
-    ws.on_upgrade(|ws| async move { run_ws(client, ws).await })
+pub async fn handle_sql_ws(
+    State(state): State<WsState>,
+    ws: WebSocketUpgrade,
+) -> impl IntoResponse {
+    ws.on_upgrade(|ws| async move { run_ws(&state, ws).await })
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+//#[serde(untagged)]
+pub struct WebSocketAuth {
+    pub user_id: String,
+    pub password: String,
+}
+
+async fn init_ws(
+    WsState {
+        frontegg,
+        adapter_client,
+    }: &WsState,
+    ws: &mut WebSocket,
+) -> Result<AuthedClient, anyhow::Error> {
+    let init_msg = ws.recv().await.ok_or_else(|| anyhow::anyhow!("closed"))??;
```

Comment:
I'm a teensy bit worried that this could cause a resource leak attack if an attacker opens up a bunch of ws connections and then does nothing. Maybe not a concern now, but we could consider implementing some form of timeout here where initial connection to valid auth takes some limited amount of time.